src/dataset-builder/TestConstants.ts

```diff
@@ -98,19 +103,124 @@ export const dummyConcepts = [
-
   { id: 300, name: 'Observation', count: 100, hasChildren: true },
   { id: 301, name: 'Blood Pressure', count: 100, hasChildren: false },
   { id: 302, name: 'Weight', count: 100, hasChildren: false },
   { id: 303, name: 'Height', count: 100, hasChildren: false },
+  { id: 400, name: 'Carcinoma of lung parenchyma', count: 100, hasChildren: true },
+  { id: 401, name: 'Squamous cell carcinoma of lung', count: 100, hasChildren: true },
+  { id: 402, name: 'Non-small cell lung cancer', count: 100, hasChildren: true },
+  { id: 403, name: 'Epidermal growth factor receptor negative ...', count: 100, hasChildren: false },
+  { id: 404, name: 'Non-small cell lung cancer with mutation in epidermal..', count: 100, hasChildren: true },
+  { id: 405, name: 'Non-small cell cancer of lung biopsy..', count: 100, hasChildren: false },
+  { id: 406, name: 'Non-small cell cancer of lung lymph node..', count: 100, hasChildren: false },
+  { id: 407, name: 'Small cell lung cancer', count: 100, hasChildren: true },
+  { id: 408, name: 'Lung Parenchcyma', count: 100, hasChildren: false },
 ];
 
 export const dummyGetConceptForId = (id: number): Concept => {
   return _.find({ id }, dummyConcepts)!;
 };
+
+export const dummyHierarchy = [
+  {
+    id: 406,
+    concept: dummyGetConceptForId(406),
+    children: [],
+    parent: 404,
+  },
+  {
+    id: 405,
+    concept: dummyGetConceptForId(405),
+    children: [],
+    parent: 404,
+  },
+  {
+    id: 404,
+    concept: dummyGetConceptForId(404),
+    children: [405, 406],
+    parent: 401,
+  },
+  {
+    id: 403,
+    concept: dummyGetConceptForId(403),
+    children: [],
+    parent: 401,
+  },
+  {
+    id: 407,
+    concept: dummyGetConceptForId(407),
+    children: [],
+    parent: 402,
+  },
+  {
+    id: 402,
+    concept: dummyGetConceptForId(402),
+    children: [407],
+    parent: 401,
+  },
+  {
+    id: 401,
+    concept: dummyGetConceptForId(401),
+    children: [402, 403, 404],
+    parent: 400,
+  },
+  {
+    id: 400,
+    concept: dummyGetConceptForId(400),
+    children: [401],
+    parent: 100,
+  },
+  {
+    id: 408,
+    concept: dummyGetConceptForId(408),
+    children: [],
+    parent: 100,
+  },
+  {
+    id: 100,
+    concept: dummyGetConceptForId(100),
+    children: [400, 408],
+  },
+];
+
+export const dummyGetNodeFromHierarchy = (id: number): ConceptNode => {
+  return _.find({ id }, dummyHierarchy)!;
+};
+
+export const getHierarchyMap = (selectedConceptID: number): Map<number, Concept[]> => {
+  const hierarchyMap = new Map<number, Concept[]>();
+  const populateHierarchyMap = (node: ConceptNode) => {
+    if (node.concept.hasChildren) {
+      const children = node.children;
+      const childrenObjects = _.map((childID) => dummyGetConceptForId(childID), children);
+      hierarchyMap.set(node.concept.id, childrenObjects);
+    }
+    if (node.parent) {
+      populateHierarchyMap(dummyGetNodeFromHierarchy(node.parent));
+    }
+  };
+
+  // get the selectedConceptNode
```

Comment:
not necessary, just an idea- selected is a bit ambiguous due to the cart. I see elsewhere you use opened, maybe just in the comment you could clarify what you mean here

Reply:
I'll change to openedConcept! Great Call!